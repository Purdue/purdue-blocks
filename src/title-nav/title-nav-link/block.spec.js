// title-nav test
jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    getAllBlocks,
    selectBlockByClientId
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickCheckbox,
} from '../../test-helpers'

const block = {blockTitle: 'Titled Navigation', blockName: 'purdue-blocks/title-nav'}

describe( 'ðŸ”¬ Titled Navigation Link Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should exist at minimum one.', async () => {
        await blockStartup(block)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$('[data-type="purdue-blocks/title-nav-link"]')).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    test( 'ðŸ”Ž Current page checkbox should add correct style.', async () => {
        await blockStartup(block)

        // select title-nav-link
        const inner = [...await getAllBlocks()][0].innerBlocks
        await selectBlockByClientId(inner.clientId)

        // force open block settings
        await clickElementByText('button', 'Block')

        // click title-nav-link to make sure block is focused and the settings appear in sidebar
        const x = await page.$( '[data-type="purdue-blocks/title-nav-link"] .block-editor-rich-text__editable' )
        await x.click()

        // click the checkbox to show button controls
        await clickCheckbox('Is this the current page?')

        const editedContent = await getEditedPostContent()
        
        expect( editedContent.includes(`wp:purdue-blocks/title-nav-link {"isCurrent":true}`)).toBe(true)
        expect( await getEditedPostContent() ).toMatchSnapshot()
    })

    test( 'ðŸ”Ž Richtext editor should update output correctly.', async () => {
        await blockStartup(block)

        const typeString = 'Text'

        // select title-nav-link
        const inner = [...await getAllBlocks()][0].innerBlocks
        await selectBlockByClientId(inner.clientId)

        // force open block settings
        await clickElementByText('button', 'Block')

        // click title-nav-link to make sure block is focused and the settings appear in sidebar
        const x = await page.$( '[data-type="purdue-blocks/title-nav-link"] .block-editor-rich-text__editable' )
        await x.click()

        await page.keyboard.type(typeString, {delay: 1})

        const editedContent = await getEditedPostContent()
        
        expect( editedContent.includes(`<li class="">${typeString}</li>`)).toBe(true)
        expect( await getEditedPostContent() ).toMatchSnapshot()
    })
})