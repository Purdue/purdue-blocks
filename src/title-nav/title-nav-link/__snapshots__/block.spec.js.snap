// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titled Navigation Link Block Block should exist at minimum one. 1`] = `
"<!-- wp:purdue-blocks/title-nav -->
<section class=\\"section pu-title-nav \\"><h2 class=\\"pu-title-nav__title\\"></h2><ul class=\\"pu-title-nav__menu\\"><!-- wp:purdue-blocks/title-nav-link -->
<li class=\\"\\"></li>
<!-- /wp:purdue-blocks/title-nav-link --></ul></section>
<!-- /wp:purdue-blocks/title-nav -->"
`;

exports[`Titled Navigation Link Block Current page checkbox should add correct style. 1`] = `
"<!-- wp:purdue-blocks/title-nav -->
<section class=\\"section pu-title-nav \\"><h2 class=\\"pu-title-nav__title\\"></h2><ul class=\\"pu-title-nav__menu\\"><!-- wp:purdue-blocks/title-nav-link {\\"isCurrent\\":true} -->
<li class=\\"isCurrent\\"></li>
<!-- /wp:purdue-blocks/title-nav-link --></ul></section>
<!-- /wp:purdue-blocks/title-nav -->"
`;

exports[`Titled Navigation Link Block Richtext editor should update output correctly. 1`] = `
"<!-- wp:purdue-blocks/title-nav -->
<section class=\\"section pu-title-nav \\"><h2 class=\\"pu-title-nav__title\\"></h2><ul class=\\"pu-title-nav__menu\\"><!-- wp:purdue-blocks/title-nav-link -->
<li class=\\"\\">Text</li>
<!-- /wp:purdue-blocks/title-nav-link --></ul></section>
<!-- /wp:purdue-blocks/title-nav -->"
`;
